### PR TITLE
Add llvm-openmp to host deps on osx

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

llvm-openmp should be pinned to 14. However, for some reason the llvm-openmp build version (not host version) is used for the version pin at the moment. This should resolve this issue.

Issue originally encountered in https://github.com/conda-forge/norse-feedstock/pull/11 where:
```
package pytorch-1.13.0-cpu_py310h09eaf1a_0 requires llvm-openmp >=15.0.6, but none of the providers can be installed
```

Both osx-64 and osx-arm64 have implicitly already pulled in llvm-openmp in the host deps in previous builds.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
Do we really not care at all if the tests pass? Why do we bother running them in the CI then anyhow?

https://github.com/conda-forge/pytorch-cpu-feedstock/blob/2be0b38024b3b5601fcefce40596fc2a5fce4ab7/recipe/meta.yaml#L136

**Comment by hmaarrfk** [comment]
I think it is just that pytorch is so heavy. It is hard to get them to pass on limited resources.

**Comment by Tobias-Fischer** [comment]
Ok that makes sense

**Comment by hmaarrfk** [comment]
i mean, i tried to debug them a while back. it is just very hard to disentangle:

1. Limited compute power.
2. Numerical error
3. Buggy code
4. Using newer versions of libraries that introduce more numerical error

We did it for scipy and numpy. You can try to run the tests. That might be good task to tackle to better understand how we are doing.

I haven't heard of too many people complaining of segfaults, so i'm crossing my fingers that we are in a good place.

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 MACOSX_SDK_VERSION:         # [osx]
-  - 12.3                    # [osx]
+  - 11.3                    # [osx]
```
Comment (hmaarrfk) [inline recipe/conda_build_config.yaml:2]: Why? I think this was useful to ensure that certain hardware optimizations were available.

**Comment by Tobias-Fischer** [comment]
Because CI fails .. It can't download the SDK image.

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
I seem to be missing something .. any clues @ngam @hmaarrfk?

```
+++ '[' -f /Users/runner/work/1/s/.ci_support/osx_64_numpy1.20python3.8.____cpython.yaml ']'
++++ cat /Users/runner/work/1/s/.ci_support/osx_64_numpy1.20python3.8.____cpython.yaml
++++ shyaml get-value MACOSX_SDK_VERSION.0 0
+++ export MACOSX_SDK_VERSION=12.3
+++ MACOSX_SDK_VERSION=12.3
++++ cat /Users/runner/work/1/s/.ci_support/osx_64_numpy1.20python3.8.____cpython.yaml
++++ shyaml get-value WITH_LATEST_OSX_SDK.0 0
+++ export WITH_LATEST_OSX_SDK=0
+++ WITH_LATEST_OSX_SDK=0
+++ [[ 0 != \0 ]]
+++ [[ 12.3 == \0 ]]
+++ export CONDA_BUILD_SYSROOT=/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk
+++ CONDA_BUILD_SYSROOT=/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk
+++ [[ ! -d /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk ]]
+++ echo 'Downloading 12.3 sdk'
Downloading 12.3 sdk
+++ curl -L -O https://github.com/phracker/MacOSX-SDKs/releases/download/11.3/MacOSX12.3.sdk.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100     9  100     9    0     0     21      0 --:--:-- --:--:-- --:--:--    21
```

It tries to download the 12.3 SDK from https://github.com/phracker/MacOSX-SDKs/ but it is not available. Are there any other variables I need to set?

**Comment by hmaarrfk** [comment]
It may be that we had a workaround for this in conda-smithy.

Could you also add:
```
skip: true    # [linux]
skip: true    # [py!=39]
```
during the debugging process

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
I feel like they pulled something
https://github.com/phracker/MacOSX-SDKs/pull/36#issuecomment-959986197

**Comment by Tobias-Fischer** [comment]
Is there anything we can do here in the meantime to push this PR forward?

**Comment by Tobias-Fischer** [comment]
Good point - I am reasonably confused now.

From the 1.13.1 build logs on osx-64 python 3.10 (https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=642018&view=logs&jobId=c35410bf-a820-56fd-e231-d3cd4bead63c&j=a8dee1cf-d17f-5cb0-b53f-4a93dfdbfb0c&t=929f6101-935e-5679-3be5-15b073490bac):
```
INFO :: The inputs making up the hashes for the built packages are as follows:
{
  "pytorch-1.13.1-cpu_py310h09eaf1a": {
    "recipe": {
      "CONDA_BUILD_SYSROOT": "/Applications/Xcode_14.0.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk",
      "c_compiler": "clang",
      "c_compiler_version": "13",
      "channel_targets": "conda-forge main",
      "cuda_compiler_version": "None",
      "cxx_compiler": "clangxx",
      "cxx_compiler_version": "13",
      "libprotobuf": "3.21",
      "llvm_openmp": "14",
      "mkl": "2022",
      "numpy": "1.21",
      "python": "3.10.* *_cpython",
      "target_platform": "osx-64"
    }
  },
  "pytorch-cpu-1.13.1-cpu_py310h439f030": {
    "recipe": {
      "channel_targets": "conda-forge main",
      "cuda_compiler_version": "None",
      "pytorch": "1.12",
      "target_platform": "osx-64"
    }
  }
}
```

But then when trying to build norse (https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=650727&view=logs&jobId=15f8dc28-9f6f-5e3a-5b3c-2b169071e5be&j=15f8dc28-9f6f-5e3a-5b3c-2b169071e5be&t=63fa899e-c160-5185-75c4-c5fd30c90374):
```
package pytorch-1.13.1-cpu_py310h09eaf1a_0 requires llvm-openmp >=15.0.7, but none of the providers can be installed
```

I have no idea what's going on there?!?

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
Many thanks for checking @ngam - sounds like this will do the job :). Is there anything else I can do here?

**Pull request closed without merging**